cardInfoAPI-Level1.yaml

```diff
@@ -0,0 +1,1402 @@
+        The API consumer is expected to have a non-card related identifier for the customer for whom data should be returned (e.g. card account number, bank account number).
+      tags:
+        - Cards
+      parameters:
+        - in: query
+          name: contract_reference
+          required: false
+          schema:
+            $ref: '#/components/schemas/ContractReference'
+          description: Retrieves all cards related to a contract. Value has to be a functional reference to the contract that can be known by the end customer.
+        - in: query
+          name: person_reference
+          required: false
+          schema:
+            $ref: '#/components/schemas/PersonReference'
+          description: Retrieves all cards related to an individual person/cardholder. Value has to be a functional reference to the person that can be known by the end customer.
+        - in: query
+          name: card_status
+          required: false
+          schema:
+            $ref: '#/components/schemas/CardStatus'
+          description: Retrieves all cards in a certain status. Only one status can be filtered at a time.
+        - in: query
+          name: card_account_reference
+          required: false
+          schema:
+            $ref: '#/components/schemas/CardAccountReference'
+          description: Retrieves all cards related to a card account (not bank account). Value has to be a functional reference to the account that can be known by the end customer.
+        - in: query
+          name: bank_account_reference
+          required: false
+          schema:
+            $ref: '#/components/schemas/BankAccountReference'
+          description: Retrieves all cards related to a bank account (not card account). Value has to be a functional reference to the account that can be known by the end customer and is usually an IBAN.
+        - $ref: '#/components/parameters/limit'
+        - $ref: '#/components/parameters/offset'
+        - $ref: '#/components/parameters/client_in_header'
+        - $ref: '#/components/parameters/correlation_in_header'
+        - $ref: '#/components/parameters/agent_in_header'
+      responses:
+        '200':
+          description: Payment cards were found which match the query parameters. If no cards matched the parameters or access rights are missing, an empty list is returned.
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/X-Correlation-ID'
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  cardList:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/CardDetails'
+                  _links:
+                    $ref: '#/components/schemas/LinksPagination'
+        '400':
+          $ref: '#/components/responses/standard400'
+        '500':
+          $ref: '#/components/responses/standard500'
```

Comment:
HTTP 500 could be considered as an unexpected status code. The consumer application has to be prepared for the unexpected case with a fallback.

There is no need to model all possible status codes. Otherwise we would need to provide many other status codes as well like 502 Bad Gateway, 504 Timeout, 428 Too many request and many more

Reply:
We are not sure how to interpret your feedback. Do you confirm our choice of using a generic 500? Or do you consider it unnecessary because an API consumer anyway must be able to handle unexpected behavior?

Reply:
We will keep the generic 500 response code to make the response codes that can occur explicit. The API user should assume that no other response codes will occur than those included in the specification.